In [1]:
import pandas as pd

df = pd.read_csv('../data/processed/telco_churn_cleaned.csv')
df.shape

(7043, 24)

In [2]:
X = df.drop('Churn', axis=1)
y = df['Churn']

In [3]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

In [4]:
X_train.shape, X_test.shape, y_train.shape, y_test.shape

((5634, 23), (1409, 23), (5634,), (1409,))

In [5]:
y_train.mean(), y_test.mean()

(np.float64(0.2653532126375577), np.float64(0.2654364797728886))

## Train/Test Split

Split the cleaned dataset into training (80%) and test (20%) sets using `train_test_split`, with:
- `random_state=42` for reproducibility if it gets rerun
- `stratify=y` to preserve the ~26.5% churn rate in both sets, since the target is imbalanced

Verified the split: 5634 training rows / 1409 test rows, with matching churn proportions 
(26.54% train, 26.54% test) confirming stratification worked correctly.

Next: train a baseline Logistic Regression model on `X_train`/`y_train`, evaluate on `X_test`/`y_test`.